staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -204,12 +205,13 @@ type LeaderElector struct {
 func (le *LeaderElector) Run(ctx context.Context) {
 	defer runtime.HandleCrash()
 	defer le.config.Callbacks.OnStoppedLeading()
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I vaguely remember some issues related to the context killing the leader elector when it shouldn't ...  I do not know if the position of the current context is in this place because of that, we should double check